lib/coverage/enabled.ml

```diff
@@ -0,0 +1,117 @@
+(** This module sets up the env variable for bisect_ppx which describes
+ *  where log files are written.
+ *)
+
+module D=Debug.Make(struct let name="coverage" end)
+let prefix = "org.xen.xapi.coverage"
+
+module Bisect = struct
+  let dump jobid =
+    let bisect_prefix = Unix.getenv "BISECT_FILE" in
+    (* dump coverage information in same location as it would normally
+       get dumped on exit, except also embed the jobid to make it easier to group.
+       Relies on [open_temp_file] generating a unique filename given a prefix/suffix to
+       avoid clashes with filenames created atexit by bisect itself. *)
+    let tmp, ch = Filename.open_temp_file
+                    ~temp_dir:(Filename.dirname bisect_prefix)
+                    (Filename.basename bisect_prefix)
+                    (Printf.sprintf ".%Ld.out" jobid) in
+    try
+      Bisect.Runtime.dump_counters_exn ch;
+      D.debug "Saved coverage data to %s" tmp;
+      close_out_noerr ch;
+      (* Keep file - will be collected by XenRT *)
+      tmp
+    with e ->
+      Sys.remove tmp;
+      D.warn "Failed to save coverage: %s" (Printexc.to_string e);
+      raise e
+
+  let reset () =
+    Bisect.Runtime.reset_counters ();
+    D.debug "Coverage counters reset";
+    "" (* no file produced *)
+
+  let init_env name =
+    let (//)    = Filename.concat in
+    let tmpdir  = Filename.get_temp_dir_name () in
+    try 
+      ignore (Sys.getenv "BISECT_FILE") 
+    with Not_found ->
+      Unix.putenv "BISECT_FILE" (tmpdir // Printf.sprintf "bisect-%s-" name)
+
+  let process body =
+    match Stringext.split ~on:' ' body with
+    | ["reset"] -> reset ()
```

Comment:
I think if we have to make `process` return a string regardless of `body`then it would feel a bit better to have this line as
`  | ["reset"] -> reset (); ""`
rather than having the `""` at the end of `reset` itself.